tests/registry_test.go

```diff
@@ -0,0 +1,162 @@
+				sess, err := cmd.Start("deis registry:unset --app=%s munkafolyamat", &user, app.Name)
+				// TODO: should unsetting a bogus registry info return 0 (success?)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, app.Name))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Specify("that user can set a valid registry information", func() {
+				sess, err := cmd.Start("deis registry:set --app=%s username=bob", &user, app.Name)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).Should(Say(`username\s+bob`))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				sess, err = cmd.Start("deis registry:list --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).Should(Say(`username\s+bob`))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Context("and registry information has already been added to the app", func() {
+
+				BeforeEach(func() {
+					sess, err := cmd.Start("deis registry:set --app=%s username=bob", &user, app.Name)
+					Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).Should(Say(`username\s+bob`))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+				})
+
+				Specify("that user can unset that registry information from that app", func() {
+					sess, err := cmd.Start("deis registry:unset --app=%s username", &user, app.Name)
+					Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).ShouldNot(Say(`username\s+bob`))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = cmd.Start("deis registry:list --app=%s", &user, app.Name)
+					Eventually(sess).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).ShouldNot(Say(`username\s+bob`))
+					Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, app.Name))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+				})
+
+			})
+
+			Specify("that user can not deploy due to lack of credentials", func() {
+				// do an unsuccessful deploy
+				image := "quay.io/deisci/e2e-private-registry-test"
+				sess, err := cmd.Start("deis pull --app=%s %s", &user, app.Name, image)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Say("Creating build..."))
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Exit(1))
+				time.Sleep(10 * time.Second)
+			})
+
+			Specify("that user can deploy with registry credentials", func() {
+				sess, err := cmd.Start("deis registry:set --app=%s username=deisci+e2e_registry password=%s", &user, app.Name, os.Getenv("QUAY_ROBOT_ACCOUNT_TOKEN"))
```

Comment:
That is in the Jenkins secret storage, but that's a fair point, I'm adding a real token here since it is fairly innocent access, read-only to a repo with only a renamed `example-go` in it